pkg/compact/compact.go

```diff
@@ -337,6 +353,105 @@ func (c *Syncer) GarbageCollect(ctx context.Context) error {
 	return nil
 }
 
+// IndexCache creates index cache file for blocks from the bucket if blocks don't have index cache.
+func (c *Syncer) IndexCache(ctx context.Context, compactDir string) error {
```

Comment:
This is definitely outside of syncer logic. Let's create another component, use syncer output (groups) to iterate and fill missing index caches instead. Plus we might need another one after/during compaction. Let's leave syncing as it's already quite complex. What do you think?

Reply:
I would even be more strict:

Add index cache file ONLY on compaction. As this is weak now as we have lots of alreafy compacted blocks, we might  want to add something temporarily as here, but ONLY for blocks with compaction level > 1 no point to do that on compaction = 1 (fresh blocks). Store will already recreate those for them.

Reply:
>Add index cache file ONLY on compaction. As this is weak now as we have lots of alreafy compacted blocks, we might want to add something temporarily as here, but ONLY for blocks with compaction level > 1 no point to do that on compaction = 1 (fresh blocks). Store will already recreate those for them.

this is indeed a temporary solution, as you can see, it didn't process index files from sidecar.